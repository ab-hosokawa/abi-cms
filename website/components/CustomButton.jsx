import { Button } from 'flowbite-react';

/**
 * カスタマイズ可能なテキストおよび追加プロパティを使用して、Button要素をレンダリングする関数コンポーネント。
 *
 * @param {Object} props - コンポーネントに渡されるプロパティ。
 * @param {string} props.text - ボタン内に表示されるテキスト。
 * @param {...Object} props.otherProps - Buttonコンポーネントに渡されるその他のプロパティ。
 * @return {JSX.Element} 指定されたテキストとプロパティを持つButtonコンポーネント。
 */
export default function CustomButton({ text, ...props }) {
  return <Button {...props}>{text}</Button>;
}
