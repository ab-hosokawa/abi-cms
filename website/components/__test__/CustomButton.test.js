// CustomButton.test.js
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CustomButton from 'components/atoms/CustomButton.jsx';

describe('CustomButtonコンポーネント', () => {
  // 基本的なレンダリングテスト
  test('ボタンが正しくレンダリングされる', () => {
    render(<CustomButton text={'Default'} type={'button'} />);
    const buttonElement = screen.getByRole('button', { name: 'Default' });
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveAttribute('type', 'button');
  });

  // スタイルのテスト
  test('ボタンが正しいスタイルクラスを持っている', () => {
    render(<CustomButton type={'button'} className={'text-white bg-blue-700 rounded-lg'} />);
    const buttonElement = screen.getByRole('button');

    // 主要なTailwindクラスが適用されているかを確認
    expect(buttonElement).toHaveClass('text-white');
    expect(buttonElement).toHaveClass('bg-blue-700');
    expect(buttonElement).toHaveClass('rounded-lg');
  });

  // クリックイベントのテスト
  test('ボタンがクリック可能である', async () => {
    const handleClick = jest.fn();
    render(<CustomButton onClick={handleClick} />);

    const buttonElement = screen.getByRole('button');
    await userEvent.click(buttonElement);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  // プロップスのテスト
  test('子要素としてのテキストを正しく表示する', () => {
    render(<CustomButton text={'カスタムテキスト'} />);
    const buttonElement = screen.getByText('カスタムテキスト');
    expect(buttonElement).toBeInTheDocument();
  });

  // disabledプロパティのテスト
  test('disabled属性が正しく機能する', () => {
    render(<CustomButton disabled={true} />);

    const buttonElement = screen.getByRole('button');
    expect(buttonElement).toBeDisabled();
  });

  // スナップショットテスト
  test('スナップショットが一致する', () => {
    const { container } = render(<CustomButton />);
    expect(container).toMatchSnapshot();
  });
});
