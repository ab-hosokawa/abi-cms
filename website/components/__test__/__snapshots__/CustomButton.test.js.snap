// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButtonコンポーネント スナップショットが一致する 1`] = `
<div>
  <button
    class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-cyan-700 text-white focus:ring-4 focus:ring-cyan-300 enabled:hover:bg-cyan-800 dark:bg-cyan-600 dark:focus:ring-cyan-800 dark:enabled:hover:bg-cyan-700 rounded-lg"
    type="button"
  >
    <span
      class="flex items-stretch transition-all duration-200 rounded-md px-4 py-2 text-sm"
    >
      <span
        class="ml-2 inline-flex h-4 w-4 items-center justify-center rounded-full bg-cyan-200 text-xs font-semibold text-cyan-800"
        data-testid="flowbite-button-label"
      />
    </span>
  </button>
</div>
`;
